File: 1768-merge-strings-alternately/1768-merge-strings-alternately.java
class Solution {
    public String mergeAlternately(String word1, String word2) {
        StringBuilder m = new StringBuilder();
        int i = 0;
        int j = 0;
        
        while(i < word1.length() && j < word2.length()){
            m.append(word1.charAt(i++));
       m.append(word2.charAt(j++));
        }
        while(i < word1.length()){
            m.append(word1.charAt(i++));
            
        }
        while(j < word2.length()){
m.append(word2.charAt(j++));
        }
        return m.toString();
    }
    
    
}

File: 1768-merge-strings-alternately/1768-merge-strings-alternately.py
class Solution:
    def mergeAlternately(self, word1: str, word2: str) -> str:
        m = len(word1)
        n = len(word2)
        i=j =0
        ans = []
        while i < m or j < n:
            if i < m :
                ans.append(word1[i])
                i +=1
            if j < n :
                ans.append(word2[j])
                j +=1
                
        return ''.join(ans)
        
        
        